compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2713,6 +2714,42 @@ private void checkConstantAssignment(BLangExpression varRef, AnalyzerData data)
         }
     }
 
+    private boolean isFiniteWithNilMember(BType bType) {
+        if (!(bType.getKind() == TypeKind.FINITE)) {
+            return false;
+        }
+        BFiniteType finiteType = (BFiniteType) bType;
+        Set<BLangExpression> finiteMembers = finiteType.getValueSpace();
+        for (BLangExpression finiteMemberType : finiteMembers) {
+            if (finiteMemberType.getBType() != null && finiteMemberType.getBType().tag != TypeTags.NIL) {
```

Comment:
Can extract `finiteMemberType.getBType()` out to a variable if the null check is needed.